modules/oe_content_featured_media_field/tests/src/Kernel/FeaturedMediaFieldTest.php

```diff
@@ -0,0 +1,240 @@
+ * @group oe_content_featured_media_field
+ */
+class FeaturedMediaFieldTest extends EntityKernelTestBase {
+
+  /**
+   * A field storage to use in this test class.
+   *
+   * @var \Drupal\field\Entity\FieldStorageConfig
+   */
+  protected $fieldStorage;
+
+  /**
+   * The field used in this test class.
+   *
+   * @var \Drupal\field\Entity\FieldConfig
+   */
+  protected $field;
+
+  /**
+   * The media entity used in this test class.
+   *
+   * @var \Drupal\media\Entity\Media
+   */
+  protected $mediaEntity;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'field',
+    'node',
+    'media',
+    'file',
+    'image',
+    'oe_media',
+    'oe_content_featured_media_field',
+    'system',
+    'text',
+    'user',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->installSchema('file', 'file_usage');
+    $this->installEntitySchema('media');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('file');
+    $this->installConfig([
+      'media',
+      'image',
+      'file',
+      'system',
+      'oe_media',
+      'oe_content_featured_media_field',
+    ]);
+
+    // Create an account with permission and set it as current user.
+    $account = $this->createUser(['view media']);
+    $this->container->get('current_user')->setAccount($account);
+
+    // Create an image file.
+    $this->container->get('file_system')->copy($this->root . '/core/misc/druplicon.png', 'public://example.jpg');
+    $image = File::create(['uri' => 'public://example.jpg']);
+    $image->save();
+
+    // Create a media entity of image media type.
+    $this->mediaEntity = Media::create([
+      'bundle' => 'image',
+      'name' => 'Test image',
+      'field_media_image' => [
+        [
+          'target_id' => $image->id(),
+          'alt' => 'default alt',
+          'title' => 'default title',
+        ],
+      ],
+    ]);
+    $this->mediaEntity->save();
+
+    // Create a content type.
+    $type = NodeType::create(['name' => 'Test content type', 'type' => 'test_ct']);
+    $type->save();
+
+    $this->fieldStorage = FieldStorageConfig::create([
+      'field_name' => 'featured_media_field',
+      'entity_type' => 'node',
+      'type' => 'oe_featured_media',
+      'cardinality' => 1,
+      'entity_types' => ['node'],
+    ]);
+    $this->fieldStorage->save();
+
+    $this->field = FieldConfig::create([
+      'label' => 'Featured media field',
+      'field_name' => 'featured_media_field',
+      'entity_type' => 'node',
+      'bundle' => 'test_ct',
+      'settings' => [
+        'handler' => 'default:media',
+        'handler_settings' => [
+          'target_bundles' => [
+            'image' => 'image',
+          ],
+        ],
+        'sort' => [
+          'field' => '_none',
+        ],
+        'auto_create' => '0',
+      ],
+      'required' => FALSE,
+    ]);
+    $this->field->save();
```

Comment:
Oh almost all of the code of the field creation is the same as the previous test. Let's create a trait that allows to create such a field, getting a few parameters (entity type, bundle, field name, and such).

Reply:
Good idea. But I'm not sure because the field settings are slightly different in each test. In the end I would end up with a method where I need to feed most of the field attributes as argument.

Reply:
Oh in that case ignore my remark 😃